src/h2n_fetcher.erl

```diff
@@ -0,0 +1,392 @@
+%%% ---------------------------------------------------------------------------
+%%% @doc
+%%%
+%%% This module fetches information from `Hex.pm` using its restful
+%%% API. *Note* The `Hex.pm` api is rate limited. Roughly 100 requests
+%%% in 60 seconds is The maximum allowed. We add a limit here of 1
+%%% request every second to make sure we don't go over that. If any
+%%% call Fails here, the entire process fails.
+%%% ---------------------------------------------------------------------------
+-module(h2n_fetcher).
+
+%%
+%% API
+%%
+-export([update_with_information_from_hex_pm/2]).
+
+%%
+%% Imports
+%%
+-include("hex2nix.hrl").
+
+%%
+%% Types
+%%
+-export_type([license/0
+             , link/0
+             , position/0
+             , dep_desc/0
+             , sha/0]).
+
+-type license() :: binary().
+-type description() :: binary() | none.
+-type link() :: binary() | no_source.
+-type position() :: root | historical.
+-type dep_desc() :: #dep_desc{}.
+-type sha() :: binary(). %% A sha 256 hash
+
+%%
+%% Definitions
+%%
+
+-define(CACHE_FILE, ".hex2nix.cache").
```

Comment:
I do not see a way to invalidate this cache anywhere. Maybe add it to "make clean" or "make clean-cache", or add additional argument to the script itself to invalidate it.